src/main/scala/tech/cryptonomic/conseil/process/AccountsTaggingProcess.scala

```diff
@@ -0,0 +1,157 @@
+      * @param fromLevel the [excluded] lowest level to load
+      * @return an effectful stream of rows, sorted with ascending block level references
+      */
+    def readOperations(fromLevel: BlockLevel): Stream[F, OperationsRow]
+
+  }
+
+  /* Actually gets all operations and flags the accounts.
+   * Returns the effectful computation that will count the number of
+   * changes to the accounts.
+   */
+  private def scan(fromLevel: BlockLevel)(implicit api: ServiceDependencies): F[Option[Int]] = {
+
+    val missingResult = Applicative[F].pure(Option.empty[Int])
+
+    /* Actually flag an account based on the operation passed, which can be a reveal or activation
+     * @return the IO action that will store the flags and return the number of changed rows, or none if
+     *         the operation was not valid
+     */
+    def flag(
+        accountRef: Option[String],
+        flagOperation: PublicKeyHash => F[Int]
+    )(
+        implicit op: OperationsRow
+    ): F[Option[Int]] =
+      accountRef match {
+        case None =>
+          logger.pureLog[F](
+            _.error("No account reference found in {}", op)
+          ) >> missingResult
+        case Some(hash) =>
+          flagOperation(PublicKeyHash(hash)).map(_.some)
+      }
+
+    /* Uppdates the passed-in reference with the new level, if higher */
+    def updateIfHigher(newLevel: BlockLevel): F[Unit] =
+      highWatermark.modify {
+        //puts the new level if it's higher than the stored one, or if there was nothing there
+        stored =>
+          stored.map(_ max newLevel).orElse(newLevel.some)
+      }
+
+    /* find sorted operations and flag the related accounts as they come by, all the while keeping track of the
+     * new block level, as read from the operations
+     */
+    api
+      .readOperations(fromLevel)
+      .evalMap { implicit operation =>
+        (operation.kind match {
+          case "reveal" =>
+            flag(operation.source, api.flagAsRevealed)
+          case "activate_account" =>
+            flag(operation.pkh, api.flagAsActive)
+          case kind =>
+            //log unexpected kind
+            logger.pureLog[F](
+              _.error(
+                """I didn't expect to process such operation kind: "{}", while flagging accounts. It should be one of {}""",
+                operation,
+                operationKinds.mkString(", ")
+              )
+            ) >> missingResult
+        }).map { flagResult =>
+          (flagResult, operation.blockLevel)
+        }
+      }
+      .evalTap {
+        //evenutually store the reference level, if needed
+        case (Some(flaggedCount), blockLevel) => updateIfHigher(blockLevel)
+        case (None, _) => Applicative[F].unit
+      }
+      .foldMap {
+        //extracts and sums all integer values (counting the actual updates)
+        case (anyFlag, _) => anyFlag.getOrElse(0)
+      }
+      .last
```

Comment:
`.compile.last` is probably what you're looking for

Reply:
leftovers of previous trials... thanks a lot